src/address.c

```diff
@@ -314,3 +314,80 @@ int verifyP2pkhAddress(char* p2pkh_pubkey, uint8_t len)
     free(dec);
     return true;
 }
+
+/**
+ * @brief This function generates a derived child key from a masterkey using
+ * a custom derived path in string format.
+ * 
+ * @param masterkey The master key from which children are derived from.
+ * @param derived_path The path to derive an address from according to BIP-44.
+ * e.g. m/44h/3h/1/1/1 representing m/44h/3h/account/ischange/index
+ * @param outaddress The derived address.
+ * @param outprivkey The boolean value used to derive either a public or 
+ * private address. 'true' for private, 'false' for public
+ * 
+ * @return 1 if a derived address was successfully generated, 0 otherwise
+ */
+int getDerivedHDAddressByPath(const char* masterkey, const char* derived_path, char* outaddress, bool outprivkey) {
+    if (!masterkey || !derived_path || !outaddress) {
+        debug_print("%s", "missing input\n");
+        return false;
+    }
+
+    /* determine if mainnet or testnet/regtest */
+    const dogecoin_chainparams* chain = chain_from_b58_prefix(masterkey);
+    bool ret = true;
+    dogecoin_hdnode node, nodenew;
+    
+    if (!dogecoin_hdnode_deserialize(masterkey, chain, &node)) {
+        ret = false;
+    }
+    
+    /* derive child key, use pubckd or privckd */
+    if (!dogecoin_hd_generate_key(&nodenew, derived_path, outprivkey ? node.private_key : node.public_key, node.chain_code, !outprivkey)) {
+        ret = false;
+    }
+
+    if (outprivkey) dogecoin_hdnode_serialize_private(&nodenew, chain, outaddress, 200);
+    else dogecoin_hdnode_serialize_public(&nodenew, chain, outaddress, 200);
+    return ret;
+}
+
+/**
+ * @brief This function generates a derived child address from a masterkey using
+ * a BIP44 standardized static, non hardened path comprised of an account, a change or
+ * receiving address and an address index.
+ * 
+ * @param masterkey The master key from which children are derived from.
+ * @param account The account that the derived address would belong to.
+ * @param ischange Boolean value representing either a change or receiving address.
+ * @param addressindex The index of the receiving/change address per account.
+ * @param outaddress The derived address.
+ * @param outprivkey The boolean value used to derive either a public or 
+ * private address. 'true' for private, 'false' for public
+ * 
+ * @return 1 if a derived address was successfully generated, 0 otherwise.
+ */
+int getDerivedHDAddress(char* masterkey, uint32_t account, bool ischange, uint32_t addressindex, char* outaddress, bool outprivkey) {
+        if (!masterkey) {
+            debug_print("%s", "no extended key\n");
+            return false;
+        }
+        char* account_str = dogecoin_char_vla(sizeof(account));
+        char* addressindex_str = dogecoin_char_vla(sizeof(addressindex));
+        char* ischange_str = ischange ? "1" : "0";
+        dogecoin_uitoa((int)account, account_str);
+        dogecoin_uitoa((int)addressindex, addressindex_str);
```

Comment:
Is it okay that this is going from unsigned to signed? The resulting string will be of the negative representation of the value if it exceeds `INT32_MAX`

Reply:
I think so and if my understanding/comment above is correct I'm not entirely sure we even need `uint32_t` as `UINT32_MAX` is 2x capacity (`4294967295`). Please let me know if I'm wrong about though! :)

Also negatives should automatically be converted to positive in the `dogecoin_uitoa` function (L544 in src/utils.c):
```
void dogecoin_uitoa(int n, char s[])
{
    int i, sign;

    if ((sign = n) < 0)  /* record sign */
        n = -n;          /* make n positive */
...
```
Does this address your concern or no?

Reply:
I will be adding some stress tests in a separate PR regardless of whether or not this acts as a haphazardly assembled bounds check.

Reply:
only caveat is that yeah, out of  x/y/z,  y = bool but only according to bip44...

Reply:
true, only needs to be `7 + 11 + 1 + 1 + 1 + 11 + 1`, not `7 + 11 + 1 + 11 + 1 + 11 + 1` as the code above has it

Reply:
It could definitely be refined. Let me know if I should do that although iirc I encountered some multi arch test failures when I was doing snprintf but it was probably a different mistake on my behalf.

Reply:
I think go with adding the comment note under a hardcoding of DERIVED_PATH_STRINGLENGTH (at 33) and then initializing as char derived_path[DERIVED_PATH_STRINGLENGTH]; is clear enough.  The 7+11+1+1+1+11+1 is too magical.

Reply:
This is reflected on L56-57 and includes that note temporarily as it will be moved to constants.h eventually.